### ISSUE TITLE
Treasure Chest: "Invalid Payment Address" error

### ISSUE THREAD

**Issue by warelock2** [PirateNetwork/pirate#109]
@CryptoForge

Problem: Any merchant using Pirate URIs will not be able to embed that URI such that the payment request details get properly sent to their customer's running instance Treasure Chest.

Client system details:

- Ubuntu 22.04
- Treasure Chest v5.8.1 Linux Appimage

Steps to reproduce:

1. Open Treasure Chest
2. Copy a valid address (ie. "zs1...")
3. Open a new window in a web browser
4. Construct a basic Pirate URI as follows: pirate:zs1...
5. Open the URI in the web browser
6. Accept the request to send the URI to Treasure Chest
7. Get an error of "Invalid Payment Address" in Treasure Chest

The code generating the error message is [here](https://github.com/PirateNetwork/pirate/blob/1d1979050336eabf62aa70f8f9eaf926bd5f2aec/src/qt/paymentserver.cpp#L512).

I first noticed this issue when trying to use the [Cryptocurrency Checkout Marketplace](https://cryptocurrencycheckout.com/marketplace). I was trying to see how their workflow is laid out when trying to use Pirate Chain to buy something through that payment processing service. You get the same error from that process, as well.

Linux command line example of how to replicate this issue (stolen from the Cryptocurrency Checkout Marketplace):

```
xdg-open "pirate:zs14l5qpyh7vf94f3vwuqmgrfm9ja8692ps7ztzl5e9xk7u8et0jd99f8k326rqt2htk42kz4y6q4p?amount=59.40111793&message=AMZNGIFT&label=RockinPricesCrypto"
```

**Comment by warelock2** [comment]
I found another issue. I run a Treasure Chest full node. When I try to use a Pirate URI, my web browser asks me if I want to send the Pirate URI to Treasure Chest. I tell it "yes" and my system tries to spin up a second instance of Treasure Chest. I then get this error in a popup from that second instance:

Pirate - Error
Cannot obtain a lock on data directory <path to home directory/.komodo/PIRATE. Pirate is probably already running.

Exactly how is this functionality supposed to work? If I'm already running a Treasure Chest full node, how is my browser supposed to get my currently running instance of Treasure Chest to open that Pirate URI? My browser's only answer to this is to just try to spin up a new copy of Treasure Chest, which will always result in a lock conflict.

Is there a way to get the registered Pirate URI mime type to get my existing Treasure Chest to handle the Pirate URI and avoid this conflict? Based on this behavior, this looks like a fundamentally broken workflow.

**Comment by warelock2** [comment]
@CryptoForge Is the issue that the Pirate URI handler code treats the Pirate Chain address as base58-encoded, rather than base32-encoded? Bech32-encoded? What encoding scheme do Pirate Chain addresses use?

**Comment by CryptoForge** [comment]
I checked this out and was able to update the encoding to use the right scheme, but then I ran into the problem of what the URI did if the wallet was already open. I need to do some more research to figure out how to get the URI to send the message to an open wallet.

**Comment by warelock2** [comment]
@CryptoForge I had previously posted what I thought was an issue with Treasure Chest, where the OS appeared to want to fire up a new instance of Treasure Chest, even though an existing instance was already running, resulting in a lock conflict. That was just a temporary glitch on my part. I just restarted Treasure Chest and the behavior went back to normal, with the existing Treasure Chest instance grabbing the event and trying to handle the Pirate URI itself, which is the expected behavior, even if it's the existing error message that I reported.

**Comment by warelock2** [comment]
> I checked this out and was able to update the encoding to use the right scheme, but then I ran into the problem of what the URI did if the wallet was already open. I need to do some more research to figure out how to get the URI to send the message to an open wallet.

@CryptoForge Just curious: What is the proper encoding scheme for Pirate Chain addresses? Bech32?

**Comment by CryptoForge** [comment]
@warelock2 This fix on the dev branch here https://github.com/PirateNetwork/pirate/commit/873eb4a2716497e30b4df9c52306151585dd397f

**Comment by warelock2** [comment]
@CryptoForge Thanks for fixing that. At least it now doesn't outright reject valid Pirate Chain Z-addresses. As discussed, the behavior is now that it pops up a dialog box for a fraction of a second, then just stops the Pirate URI handling workflow altogether.

**Comment by warelock2** [comment]
@CryptoForge  Native Pirate URI handling has been fixed in Treasure Chest v5.8.2. Closing this issue.
